src/PrettyPrinting.jl

```diff
@@ -1382,150 +1382,245 @@ end
-#
-# to
-#
-#   @attributes mutable struct bla ..
-#   ...
-#   end
-#
-# Then, in the `show` method, start with
-#   @show_name(io, obj)
-# If the user assigned a name to the object (in the REPL mainly) by doing
-# A = bla...
-# then, in the compact printing only the name "A" is printed
-# also adding
-# @show_special(io, obj)
-# allows, if present to call a different printing function for this instance
-# See FreeModule for an example
+#  Macros for fancy printing. 
 #
 ###############################################################################
 
-function set_name!(G::Any, name::String)
-   set_attribute!(G, :name => name)
+const CurrentModule = Ref(Main)
+
+function set_current_module(m)
+  CurrentModule[] = m
 end
 
-function set_name!(G)
-   s = get_attribute(G, :name)
-   s === nothing || return
-   sy = find_name(G)
-   sy === nothing && return
-   set_name!(G, string(sy))
+function get_current_module()
+  return CurrentModule[]
 end
 
-extra_name(G) = nothing
+"""
+    set_name!(obj, name::String; override::Bool=true)
 
-macro show_name(io, O)
-  return :( begin
-    local i = $(esc(io))
-    local o = $(esc(O))
-    s = get_attribute(o, :name)
-    if s === nothing
-      sy = find_name(o)
-      if sy === nothing
-        sy = extra_name(o)
-      end
-      if sy !== nothing
-        s = string(sy)
-        set_name!(o, s)
-      end
-    end
-    if s !== nothing && (get(i, :supercompact, false) || get(i, :compact, false))
-      if AbstractAlgebra.PrettyPrinting._supports_io_custom(i)
-        print(i, LowercaseOff())
-      end
-      print(i, s)
-      return
-    end
-  end )
+Sets the name of the object `obj` to `name`.
+This name is used for printing using [`AbstractAlgebra.@show_name`](@ref).
+If `override` is `false`, the name is only set if there is no name already set.
+
+This function errors if `obj` does not support attribute storage.
+"""
+function set_name!(obj, name::String; override::Bool=true)
+  override || isnothing(get_attribute(obj, :name)) || return
+  set_attribute!(obj, :name => name)
 end
 
-const CurrentModule = Ref(Main)
+"""
+    set_name!(obj; override::Bool=true)
 
-function set_current_module(m)
-  CurrentModule[] = m
-end
+Sets the name of the object `obj` to the name of a variable in global (`Main` module) namespace
+with value bound to the object `obj`, if such a variable exists (see [`AbstractAlgebra.find_name`](@ref)).
+This name is used for printing using [`AbstractAlgebra.@show_name`](@ref).
+If `override` is `false`, the name is only set if there is no name already set.
 
-function get_current_module()
-  return CurrentModule[]
+This function errors if `obj` does not support attribute storage.
+"""
+function set_name!(obj; override::Bool=true)
+  override || isnothing(get_attribute(obj, :name)) || return
+  sy = find_name(obj)
+  isnothing(sy) && return
+  set_name!(obj, string(sy); override=true)
 end
 
-function find_name(A, M = Main; all::Bool = false)
+"""
+    extra_name(obj) = nothing
```

Comment:
Did you mean
```suggestion
    extra_name(obj) -> Union{String,Nothing}
```
Or is this meant to indicate the default value? But that seems unusual, I'd rather state it in the text.